### PR TITLE
Enable decoding  only single frame from animated gif + fix quantization issue

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This PR was originally designed to simply add a `FrameDecodingMode` enumeration for decoding animated gifs but I discovered an inconsistency in quantization while I was cleaning up the codebase. 

As such I've combined the fix + tests to this PR.

### CODE REVIEW

**Comment by antonfirsov** [comment]
@JimBobSquarePants I really want to review this, but I'm being short on time now. Do we still have a few days for this, or do you want to go fast because of beta-2?

**Comment by JimBobSquarePants** [comment]
@antonfirsov I'm happy to delay beta-2 for this. The fix is an important one.

File: tests/ImageSharp.Tests/Formats/Gif/GifDecoderTests.cs [hunk 1]
```diff
@@ -79,18 +78,40 @@ public void Decode_IgnoreMetadataIsTrue_CommentsAreIgnored()
...
+        }
+
+        [Theory]
+        [WithFile(TestImages.Gif.Giphy, PixelTypes.Rgba32)]
+        public void CanDecodeAllFrames<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage(new GifDecoder { DecodingMode = FrameDecodingMode.All }))
+            {
+                Assert.True(image.Frames.Count > 1);
```
Comment (antonfirsov) [inline tests/ImageSharp.Tests/Formats/Gif/GifDecoderTests.cs]: Maybe not for this PR but ...
there are only 5 frames in this gif. By saving all of them as PNG we can introduce quite good regression testing for the gif decoder!

File: src/ImageSharp/Formats/Gif/GifEncoderCore.cs [hunk 2]
```diff
@@ -154,18 +154,14 @@ private int GetTransparentIndex<TPixel>(QuantizedImage<TPixel> quantized)
         {
             // Transparent pixels are much more likely to be found at the end of a palette
             int index = -1;
+            var trans = default(Rgba32);
             for (int i = quantized.Palette.Length - 1; i >= 0; i--)
             {
-                quantized.Palette[i].ToXyzwBytes(this.buffer, 0);
+                quantized.Palette[i].ToRgba32(ref trans);
```
Comment (antonfirsov) [inline src/ImageSharp/Formats/Gif/GifEncoderCore.cs]: Cool to see more of these dropped!

**Comment by antonfirsov** [comment]
Also, I'm copying my gitter comment here, because it's relevant:
> shouldn't we add 300x300-c.gif to our test images?

File: src/ImageSharp/Formats/Gif/GifEncoderCore.cs (hunk 2 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Gif/GifEncoderCore.cs]: Yeah, I wanna replace all of them and remove the method very soon.

File: tests/ImageSharp.Tests/Formats/Gif/GifDecoderTests.cs (hunk 1 above)
Comment (JimBobSquarePants) [reply tests/ImageSharp.Tests/Formats/Gif/GifDecoderTests.cs]: Indeed! I'll have a look

**Pull request merged**